### PR TITLE
Feat/lm service

### PR DESCRIPTION
**Pull request by blester125**
This PR is tracking the addition of a `score` function to the language models, it can give you the conditional probability of a token given a context or the probability of a sequence.

Tested with the tf word level language model, the pytorch kim and word model. The tf kim model is currently broken but I think this can be merged before that is fixed.

### CODE REVIEW

File: baseline/services.py [hunk 1]
```diff
@@ -399,6 +403,83 @@ def load(cls, bundle, **kwargs):
...
+        # scores = torch.gather(softmax_tokens, -1, values.unsqueeze(-1))
+        scores = []
+        # The last softmax is what comes after the <EOS> so we don't grab from there
+        for soft, value in zip(softmax_tokens[:, :-1], values):
+            tokens = []
+            for tok, val in zip(soft, value):
+                tokens.append(tok[val].item())
+            scores.append(tokens)
+        scores = np.array(scores)
+        logger.warning("Warning: The `joint` probability isn't well formed right now and should be used with a grain of salt")
```
Comment (dpressel) [inline baseline/services.py]: I dont think we need this line

File: layers/eight_mile/utils.py [hunk 2]
```diff
@@ -1548,3 +1548,16 @@ def mime_type(file_name: str) -> str:
...
         return "text/plain"
     return "application/w2v"
+
+
+@export
+def to_numpy(x):
+    if isinstance(x, np.ndarray):
+        return x
+    try:
+        import torch
```
Comment (dpressel) [inline layers/eight_mile/utils.py]: can actually test if its an instance of a torch.Tensor right?

Comment (blester125) [reply layers/eight_mile/utils.py]: Why do we need to test that? If you are using tf1 then your predict result is already a numpy array and is caught by the first check, if you are using tf2 and you don't have pytorch installed the import causes an error which we catch. We then call `.numpy()` to turn the `tf.Tesor` into a numpy array. If you have pytorch installed but it was a tf2 tensor the `.detach()` throws which we again catch and just us `.numpy()`. If we are using pytorch we use `.detach()` which is defined on Tensors with and without `requires_grad=True` and then we move it to `cpu` before calling `.numpy()`

I guess an explicit check (while keeping the try in case they don't have pytorch installed) is probably simpler to reason when we need to look at this function again

Comment (dpressel) [reply layers/eight_mile/utils.py]: it could be a little more generalizable in the future, and more precise about what its trying to accomplish.
